responses.py

```diff
@@ -288,9 +290,15 @@ def unbound_on_send(adapter, request, *a, **kwargs):
     def stop(self):
         self._patcher.stop()
         if self.assert_all_requests_are_fired and self._urls:
-            raise AssertionError(
+            assertion_error = AssertionError(
                 'Not all requests has been executed {0!r}'.format(
                     [(url['method'], url['url']) for url in self._urls]))
+            if self.captured_exception:
+                try:
```

Comment:
Doesn't this try-except block shadow still eat up other exceptions? Removing it, and instead just re-raising the captured exception, seems to give the correct behaviour for me. Am I missing something?